decode.go

```diff
@@ -1328,6 +1355,16 @@ func (d *decoder) parseToTime() (time.Time, bool, error) {
 	}
 
 	switch t := d.nextCBORType(); t {
+	case cborTypeByteString:
+		if d.dm.byteStringToTime == ByteStringToTimeAllowed {
+			b, _ := d.parseByteString()
+			t, err := time.Parse(time.RFC3339, string(b))
+			if err != nil {
+				return time.Time{}, false, errors.New("cbor: cannot set " + string(b) + " for time.Time: " + err.Error())
```

Comment:
We should escape control characters and non-printable characters in the error message.

```suggestion
				return time.Time{}, false, errors.New("cbor: cannot set " + strconv.Quote(string(b)) + " for time.Time: " + err.Error())
```